Makefile

```diff
@@ -1,39 +1,40 @@
 # Makefile to build and run MQTT plugin for Spin framework.
 
 .PHONY: all
-all: build_plugin package_plugin install_plugin
+all: build_plugin install_plugin
+
+.PHONY: lint
+lint:
+	@echo "Running linting check..."
+	cargo clippy --all --all-targets -- -D warnings
+	cargo fmt --all -- --check
+
+.PHONY: lint-rust-examples
+lint-rust-examples:	
+		@echo "Running linting check on example..." \
+		&& cargo clippy --manifest-path "examples/mqtt-app/Cargo.toml" -- -D warnings \
+		&& cargo fmt --manifest-path "examples/mqtt-app/Cargo.toml" -- --check \		
+	done
+
+.PHONY: lint-all
+lint-all: lint lint-rust-examples
 
 .PHONY: build_plugin
 build_plugin:
 	@echo "Building Mqtt Plugin..."
-	cargo fmt --check && cargo build --release
-
-.PHONY: package_plugin
-package_plugin:
-	@echo "Packaging Mqtt Plugin in Spin..."
-	# Copy the binary to the root directory
-	cp ./target/release/trigger-mqtt .
-	# Create a tarball
-	(tar czvf trigger-mqtt.tar.gz trigger-mqtt)
+	cargo build --release
 
 .PHONY: install_plugin
 install_plugin:
 	@echo "Installing Mqtt Plugin in Spin..."
-	$(eval SHA256 := $(shell shasum -a 256 trigger-mqtt.tar.gz | cut -d ' ' -f 1))
-	$(eval PWD := $(shell pwd))
-	@echo "Updating sha256 in plugin manfiest..."
-	sed -i 's|"sha256": ".*"|"sha256": "$(SHA256)"|' ./trigger-mqtt-local.json
-	@echo "Updating plugin file path in plugin manfiest..."
-	sed -i 's|"url": ".*"|"url": "file://$(PWD)/trigger-mqtt.tar.gz"|' ./trigger-mqtt-local.json
-	spin plugin uninstall trigger-mqtt && spin plugin upgrade --file ./trigger-mqtt-local.json --yes
-	spin plugins list --installed | grep trigger-mqtt
-
-	rm trigger-mqtt.tar.gz
-	rm trigger-mqtt
-
+	spin plugins upgrade pluginify -y
```

Comment:
Could we just specify it as a pre-requisite? I am just wondering if it is worth running `update` and `upgrade` each time because `update` does a `git pull` on the [registry](https://github.com/fermyon/spin-plugins) and I also do not expect `pluginify` to be updated too frequently. 

 I am happy with it either way (just exploring the options).

Reply:
Yeah that's an interesting/valid point, let's get this in for now and improve in future if it's pressing enough?
Ideally, we probably do want to install the latest plugin if it's updated so we do not drift, possibly can update this to check version diff and install if newer version is available but then I also think `spin plugin upgrade` may/should be doing this under the hood already? not sure if we can avoid update check and if it's that much taxing, just a thought...

Reply:
Yea agreed. It is not a pressing issue and even pull should not be too bad anytime soon. So It is good to punt on it.